odbc_fdw.c

```diff
@@ -2229,7 +2229,12 @@ odbcImportForeignSchema(ImportForeignSchemaStmt *stmt, Oid serverOid)
 		foreach(option, stmt->options)
 		{
 			DefElem *def = (DefElem *) lfirst(option);
-			appendOption(&create_statement, ++option_count == 1, def->defname, defGetString(def));
+			// options not in the CREATE FOREIGN TABLE statement will have location == -1
+			// we'll ignore them as they are defined by the SERVER or USER MAPPING, and including them here
+			// would be functional but could expose sensitive information
+			if (def->location != -1) {
```

Comment:
What are the options that we are passing here? I see postgres_fdw using only `schema_name` and `table_name`. Does odbc pass other options related to tables?

Reply:
We pass any attribute XXX that we need to set in the ODBC connection through an OPTION named `ODBC_XXX`. As a convention we pass any attribute that defines the connection in the SERVER options (but we not be strict), but what's important is that we pass attributes thats should be confidencial (password,...) in the USER MAPPING options (so regular users cannot read them). Here in options we have all the accumulated OPTIONS from the SERVER, USER MAPPING and the IMPORT, and we have been echoing all blindly to the CREATE FOREIGN TABLE. But we need only set those from the IMPORT. The `def-location` is the position of the option in the IMPORT statement text, and for those that don't come from the IMPORT statement the value is -1.

I hope I answered your question 😛